openslides_backend/action/action_handler.py

```diff
@@ -245,4 +249,6 @@ def perform_action(
             # -1: error which cannot be directly associated with a single action data
             if action.index > -1:
                 exception.action_data_error_index = action.index
+            if on_failure := action.get_on_failure(action_data):
+                self.on_failure.append(on_failure)
```

Comment:
You can't gether the `on_failure` calls here because `perform_action` is possibly called multiple times per action (see `execute_write_requests` above: If the datastore returns a `DatastoreLockedException`, the request is retried). So suppose we want to execute `poll.start`. The first try fails and adds the `on_failure` routine to the list, then the second try succeeds. Then, at the end, the `on_failure` routine is still executed and the poll is reset even though it shouldn't have been, which leads to an inconsistent state.

The easiest solution is probably to execute the `on_failure` routine here directly. So you don't need the `on_failure` list at all.

Reply:
Done.